src/pages/contact.tsx

```diff
@@ -1,13 +1,33 @@
+import { useState } from 'react'
+
+import CONTACT_DATA from '@/../mockData/CONTACT_DATA.json'
+import ContactList from '@/components/Contact/contactlist'
 import Header from '@/components/Header'
+import SearchBar from '@/components/SearchBar/SearchBar'
 
 const Contact = () => {
+  const [searchField, setSearchField] = useState('')
+
+  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const searchFieldString = event.target.value.toLocaleLowerCase()
+    setSearchField(searchFieldString)
+  }
+
+  const filteredContacts = CONTACT_DATA.filter((contact) => {
+    const full_name = contact.first_name + ' ' + contact.last_name
+    return full_name.toLocaleLowerCase().includes(searchField)
+  })
```

Comment:
If this code was to not change, it would be fine. However if additional state variables were added to the component, the filteredContacts variable would be re-assigned on every single render even if the searchField state wasn't changing. If the list of contacts was large, this would waste rendering time. 

I think a better way would be to use filteredContacts as a state instead of searchField. In the onSearchChange() function, the filteredContacts state could be updated.

Reply:
@APM246 I've changed it so that it uses filteredContacts as a state. Are you able to check if I did it right? 

Also off-topic but we have a lot of repeated use of the type "contactData" would it be better to put these types in its own file and import it where it's needed rather than repeatedly writing it?

Reply:
Yea looks great :)

Yea putting contactData in a separate file would be good so there's only 1 consistent definition of it

Reply:
@APM246 Done, thanks :)

Do you have any suggestions on how to specify the correct type for 'context' for getStaticProps under pages/contact/[id].tsx I can only get it to work with type 'any' but the GitHub checks says not to use type 'any'. Not sure if I've done this page right.

Reply:
It looks like NextJS doesn't have a declared type for the context param so you'll just have to define a custom type for it. That type declaration could go in types/types.ts (https://nextjs.org/docs/api-reference/data-fetching/get-static-props).

I think the [id].tsx looks fine, probably doesn't matter since it'll have to be changed once the API is built. Also don't forget to add the async keyword to getStaticPaths and getStaticProps